### ISSUE TITLE
Data Browser Plot & Model update  (Lead: Kay- Reviewers: Nadine)

### ISSUE THREAD

**Issue by kasemir** [ControlSystemStudio/cs-studio#774]
The data browser started with a rather simple but functional plot:
 * Each trace had a different color
 * Name of trace defaulted to PV name + units, but could be changed
 * Name of trace, in the trace's color, shown on value axis
 * PVs could be dropped onto axes or main plot area, adding the PV to that axis or a new axis
 * High contrast, full time information

It may not have been the prettiest, but totally readable and a snapshot of a plot in for example the logbook had the full information: What channels do I see? What's the time range? The day/year? What's the value of the points on a trace?

The update to the XYGraph added many configuration options. Axes no longer showed the trace names, which moved to a new legend. Fonts, axis positions, .. could be tweaked. But not all configuration options make sense, for example a logarithmic X axis that doesn't show time stamps at all. An early update got trace names back onto the axes, but now the legend duplicates the info, wasting screen spare. It is quite easy to end up with the same trace color for multiple traces on the same axis, i.e. indistinguishable traces which then end in the logbook and viewers cannot tell what the actual values are. Several people worked on the code to save all XYGraph options, and it's now quite broken: #755, #694, #632, #536, #428, #213.
Finally, both the original plot and the XYGraph perform most of their work on the UI thread, which is slow.

Plan:
Move back to a plot similar to the original one, but perform most of the plotting in background threads. Support VStatistics-type data to allow #764. Properly save and restore the plot model.

Regarding backwards compatibility, all *.plt files will still "work": PVs, axis assignments, archive data sources, basic plot modes (area or simple line, color, line width) will remain as before. XYGraph specific settings like fonts, dash-dotted line, axis-on-right, .. will likely be lost. Annotations and cursors need to remain available.

Bottom line, the result should be plainer but faster.

**Comment by kasemir** [comment]
Data browser update is fundamentally done.
For bugs in the new code, create new tickets.
